File: examples/python/env_vars.py
"""Print host-provided environment variables."""
import os

# These are set by the host via --env KEY=VALUE (CLI) or
# GuestConfig::set_env_vars() (library API).
my_var = os.environ.get("MY_VAR", "NOT_SET")
debug = os.environ.get("DEBUG", "NOT_SET")
greeting = os.environ.get("GREETING", "NOT_SET")

print(f"MY_VAR={my_var}")
print(f"DEBUG={debug}")
print(f"GREETING={greeting}")

File: examples/rust/env_vars.rs
fn main() {
    let my_var = std::env::var("MY_VAR").unwrap_or_default();
    let debug = std::env::var("DEBUG").unwrap_or_default();
    let greeting = std::env::var("GREETING").unwrap_or_default();
    println!("MY_VAR={my_var}");
    println!("DEBUG={debug}");
    println!("GREETING={greeting}");
}
